### PR TITLE
Fix recursive format

### LINKED ISSUE #17: Recursive variables issue
Variable like this may have an issue if the order isn't good:
```
first: 111
second: "{first}"
third: "{second}"
```

### PR DESCRIPTION
**Pull request by sbrunner**
fix #17

### CODE REVIEW

File: c2c/template.py [hunk 1]
```diff
@@ -89,17 +90,46 @@ def main():
 
     used_vars = read_vars(options.vars)
 
-    def format_walker(current_vars):
+    formatter = Formatter()
+    formated = []
```
Comment (asaunier) [inline c2c/template.py]: formated => formatted

File: c2c/template.py [hunk 2]
```diff
@@ -89,17 +90,46 @@ def main():
...
+                for _, attr, _, _ in attrs:
+                    if attr is not None and attr not in formated:
+                        return current_vars, 1
+                formated.append(path)
+                return current_vars.format(**used_vars), 0
+            return current_vars, 0
+
         elif isinstance(current_vars, list):
-            return [format_walker(var) for var in current_vars]
+            formateds = [
```
Comment (asaunier) [inline c2c/template.py]: formatteds

File: c2c/template.py [hunk 3]
```diff
@@ -89,17 +90,46 @@ def main():
...
+                format_walker(var, "%s[%i]" % (path, index))
+                for index, var
```
Comment (asaunier) [inline c2c/template.py]: Doesn't it fit on a single line?

Comment (sbrunner) [inline c2c/template.py]: Your right that's better :-)

**Pull request merged**